services/analyzer/sequelize-column-type-getter.js

```diff
@@ -72,11 +73,14 @@ function ColumnTypeGetter(databaseConnection, schema, allowWarning = true) {
     return `ARRAY(DataTypes.${await this.perform(innerColumnInfo, innerColumnInfo.udtName, tableName)})`;
   };
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   this.perform = async (columnInfo, columnName, tableName) => {
     const { type } = columnInfo;
     const mysqlEnumRegex = /ENUM\((.*)\)/i;
 
     switch (type) {
+      case (type === 'JSON' && !isDialect(DIALECT_MSSQL) && 'JSON'):
+        return 'JSON';
```

Comment:
[Ignorable] This syntax id not that common. However, I have no suggestion that does not require a full refactor (you mentioned the strategy pattern, but I guess it's out of scope).